### PR TITLE
Plugin installation

### LINKED ISSUE #43: Implement install button
Implement functionality for the install button in the Preferences window.

It should download the binary and execute it once placed in `~/.local/share/done/bin`

### PR DESCRIPTION
**Pull request by edfloreshz**
This PR implements plug-in installation functionality in the preferences pane.

image

Closes #43

### CODE REVIEW

File: src/application/plugin.rs [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 use serde::{Deserialize, Serialize};
+use std::io::Write;
+use std::os::unix::prelude::PermissionsExt;
+use std::path::PathBuf;
 use std::process::Command;
 use sysinfo::{ProcessExt, System, SystemExt};
 use tonic::transport::Channel;
 
-pub const PLUGINS_URL: &str = "https://raw.githubusercontent.com/done-devs/done/beta/dev.edfloreshz.Done.Plugins.json";
+pub const PLUGINS_URL: &str = "https://raw.githubusercontent.com/done-devs/done/feat-install-plugin/dev.edfloreshz.Done.Plugins.json";
```
Comment (AaronErhardt) [inline src/application/plugin.rs:16]: I think this should target the beta branch again so it continues to work after the merge.

File: src/widgets/factory/service_row.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+#[relm4::factory(pub async)]
+impl AsyncFactoryComponent for ServiceRowModel {
+	type ParentInput = PreferencesComponentInput;
+	type ParentWidget = adw::PreferencesGroup;
+	type CommandOutput = ();
+	type Input = ServiceRowInput;
+	type Output = ServiceRowOutput;
+	type Init = PluginPreferences;
+	type Widgets = ProviderWidgets;
```
Comment (AaronErhardt) [inline src/widgets/factory/service_row.rs:45]: ProviderWidgets doesn't look right to me. You can just delete the Widgets type here to let the macro name it.

File: src/widgets/factory/service_row.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+	view! {
+				#[root]
+		#[name(service)]
+				adw::ActionRow {
```
Comment (AaronErhardt) [inline src/widgets/factory/service_row.rs:50]: Another real-world use-case for relm4-format :P

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks good in general, just a few minor things should be adjusted.

File: src/widgets/factory/service_row.rs (hunk 3 above)
Comment (edfloreshz) [reply src/widgets/factory/service_row.rs]: Lol. fixed!

File: src/widgets/factory/service_row.rs (hunk 2 above)
Comment (edfloreshz) [reply src/widgets/factory/service_row.rs]: Removed from everywhere this applies to.

File: src/application/plugin.rs (hunk 1 above)
Comment (edfloreshz) [reply src/application/plugin.rs]: Fixed 🎉

**Comment by edfloreshz** [comment]
Thanks @AaronErhardt 🎉

**Comment by edfloreshz** [comment]
Close #45

**Pull request merged**